Resources/Prototypes/_NF/Entities/Objects/Vehicles/vehicles.yml

```diff
@@ -183,6 +183,166 @@
           params:
             volume: -3
 
+# Enhanced version of the vanilla Janicart that adds a bucket and mop slot.
+- type: entity
+  id: VehicleJanicartNF
+  parent: BaseVehicleRideable
+  name: janicart
+  description: The janitor's trusty steed.
+  components:
+  - type: Vehicle
+    southOver: true
+    westOver: true
+    eastOver: true
+    northOverride: -0.15
+    southOverride: 0.22
+  - type: Sprite
+    sprite: Objects/Vehicles/janicart.rsi
+    layers:
+    - state: vehicle
+      map: ["enum.VehicleVisualLayers.AutoAnimate"]
+    noRot: true
+  - type: UnpoweredFlashlight
+  - type: PointLight
+    enabled: false
+    radius: 3.5
+    softness: 2
+    mask: /Textures/Effects/LightMasks/cone.png
+    autoRot: true
+  - type: Destructible
+    thresholds:
+    - trigger:
+        !type:DamageTrigger
+        damage: 500
+      behaviors:
+        - !type:DoActsBehavior
+          acts: [ "Destruction" ]
+    - trigger:
+        !type:DamageTrigger
+        damage: 250
+      behaviors:
+      - !type:DoActsBehavior
+        acts: ["Destruction"]
+      - !type:PlaySoundBehavior
+        sound:
+          collection: MetalGlassBreak
+      - !type:EmptyAllContainersBehaviour
+      - !type:ExplodeBehavior
+      - !type:SpawnEntitiesBehavior # in future should also emit a cloud of hot gas
+        spawn:
+          VehicleJanicartDestroyed:
+            min: 1
+            max: 1
+  - type: ItemSlots
+    slots:
+      mop_slot:
+        name: janitorial-trolley-slot-component-slot-name-mop
+        whitelist:
+          tags:
+            - Mop
+        insertOnInteract: false # or it conflicts with bucket logic
+        priority: 9 # Higher than bucket slot
+      bucket_slot:
+        name: janitorial-trolley-slot-component-slot-name-bucket
+        whitelist:
+          tags:
+            - Bucket
+        insertOnInteract: false # or it also conflicts with bucket logic
+        priority: 4 # Higher than trash bag slot
+      key_slot:
+        name: vehicle-slot-component-slot-name-keys
+        whitelist:
+          requireAll: true
+          tags:
+          - VehicleKey
+          - JanicartKeys
+        insertSound:
+          path: /Audio/Effects/Vehicle/vehiclestartup.ogg
+          params:
+            volume: -3
+      trashbag_slot:
+        name: janitorial-trolley-slot-component-slot-name-trashbag
+        whitelist:
+          tags:
+            - TrashBag
+        priority: 3 # Higher than drinking priority
```

Comment:
Why not go for parity with the trolley itself, add slots for wet floor signs, anything else it has?  Hell, you might be able to reparent from both the Janicart and the janitorial trolley without much issue.

Reply:
Felt a bit weird to have _that_ much stuff tacked onto it given how sleek the vehicle design is... But eh why not. 

I'll have to try out the multi-parenting. Would be slick if it Just Worked(tm).

Reply:
Multi-parenting didn't work smoothly for me, but I added the additional slots.